packages/web-react/src/components/FileUploader/FileUploaderInput.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { RefObject, createRef } from 'react';
+import classNames from 'classnames';
+import { SpiritFileUploaderInputProps } from '../../types';
+import { useStyleProps } from '../../hooks';
+import { DEFAULT_FILE_QUEUE_LIMIT, DEFAULT_FILE_SIZE_LIMIT } from './constants';
+import { useFileUploaderStyleProps } from './useFileUploaderStyleProps';
+import { useFileUploaderInput } from './useFileUploaderInput';
+import { Icon } from '../Icon';
+
+const FileUploaderInput = (props: SpiritFileUploaderInputProps) => {
+  const {
+    children,
+    fileLimitMaxSize = DEFAULT_FILE_SIZE_LIMIT,
+    fileQueueLimit = DEFAULT_FILE_QUEUE_LIMIT,
+    helperText,
+    id,
+    isDisabled,
+    isRequired,
+    label,
+    labelText,
+    linkText,
+    onError,
+    validationMessage,
+    validationState,
+    ...restProps
+  } = props;
+
+  const hasDragAndDrop = 'draggable' in document.createElement('span');
+  const inputReference = createRef() as RefObject<HTMLInputElement>;
+  const dropZoneReference = createRef() as RefObject<HTMLDivElement>;
+
+  const { isDragging } = useFileUploaderInput({
+    dropZoneReference,
+    fileLimitMaxSize,
+    fileQueueLimit,
+    hasDragAndDrop,
+    inputReference,
+    onError,
+  });
+  const { classProps } = useFileUploaderStyleProps({
+    hasDragAndDrop,
+    isDisabled,
+    isDragging,
+    isRequired,
+    validationState,
+  });
+  const { styleProps, props: transferProps } = useStyleProps(restProps);
+
+  return (
+    <div {...transferProps} {...styleProps} className={classNames(classProps.input.root, styleProps.className)}>
+      <label htmlFor={id} className={classProps.input.label}>
+        {label}
+      </label>
+      <div ref={dropZoneReference} className={classProps.input.dropZone.root}>
+        {children || (
+          <>
+            <Icon name="upload" aria-hidden="true" />
+            <label htmlFor={id} className={classProps.input.dropZone.label}>
+              <span className={classProps.input.link}>{linkText}</span>
+              <span className={classProps.input.dropLabel}>{labelText}</span>
+            </label>
+            <div className={classProps.input.helper}>{helperText}</div>
+          </>
+        )}
+        <input type="file" ref={inputReference} id={id} className={classProps.input.input} {...restProps} />
+      </div>
+      {validationState && validationMessage && <div className={classProps.input.message}>{validationMessage}</div>}
```

Comment:
I think we will be renaming this soon in V1 to `validationText`.
```suggestion
      {validationState && validationText && <div className={classProps.input.message}>{validationMessage}</div>}
```

Reply:
done